build-system/server/new-server/transforms/modules/modules-transform.ts

```diff
@@ -18,18 +18,33 @@ import {PostHTML} from 'posthtml';
 import {URL} from 'url';
 import {isValidScript, ScriptNode} from '../utilities/script';
 import {CDNURLToLocalDistURL} from '../utilities/cdn';
+import {OptionSet} from '../utilities/option-set';
+import minimist from 'minimist';
+const argv = minimist(process.argv.slice(2));
 
 /**
  * Append a Module Script for a ScriptNode.
  * @param head
  * @param script
  */
-function appendModuleScript(head: PostHTML.Node, script: ScriptNode): void {
-  const modulePath = CDNURLToLocalDistURL(
-    new URL(script.attrs.src || ''),
-    undefined,
-    '.mjs'
-  ).toString();
+function appendModuleScript(head: PostHTML.Node, script: ScriptNode, compiled: boolean): void {
+  let modulePath;
+  if (argv.compiled || compiled) {
+    modulePath = CDNURLToLocalDistURL(
+      new URL(script.attrs.src || ''),
```

Comment:
are you able to re-use the URL instance here to be used below?

Reply:
Fixed.